.github/workflows/release-dryrun.yml

```diff
@@ -0,0 +1,30 @@
+# Copyright 2023 Terramate GmbH
+# SPDX-License-Identifier: MPL-2.0
+
+name: release-dryrun
+on:
+  pull_request:
+    paths:
+      - 'VERSION'
```

Comment:
the `release-dryrun` must run on all events because it checks if Terramate is releasable. We can't just discover that Terramate is not releasable right at the time of releasing it, this surprise could cause us a lot of headache.

Like, if someone introduces a change in the `make build` that now makes some compiler flags required, this will break the release if the `goreleaser.yml` is not updated. There are many other ways of making the release broken, then I think we should continue running it in the PR on each push. 
It's fine moving to a separate workflow though, I think that makes sense.

Reply:
Good point, `VERSION` is indeed not enough and I agree about the correctness of the release building process.
However, most PRs are not changing the release process so it seems suboptimal to check it on every change.

We could add the `Makefile`, `makefiles/**`, `.goreleaser.yml` ro reduce the likelihood of breaking the build. What other files could possibly break the release building process?

Reply:
Okay, yes maybe we can optimize to check for those files. Additionally you also need to check for `hack/release/Dockerfile`.

The only other of breaking it, I guess, is if we move the `main.go` location of `cmd/terramate` or `cmd/terramate-ls`, but this is very unlikely.

Go ahead 👍